### ISSUE TITLE
Image size

### ISSUE THREAD

**Issue by hungdev** [zoontek/react-native-bootsplash#402]
### Bug summary

Hi @zoontek
Thank for your library.
what is input image size. ?pixel x ?pixel
I used the default command without logo-width: yarn react-native generate-bootsplash assets/bootsplash_logo_original.png
But i got the small image on the splash screen. And after that i use --logo-width=300, and it got the error: ` Logo width can't be superior to 288dp as it will be cropped on Android. Exiting…`
Then i changed to `npx react-native generate-bootsplash ./assets/images/splash.png \ --background-color=FFFFFF \
  --logo-width=288`
then i got the message:
⚠️   As logo width is superior to 192dp, it might be cropped on Android.

    Android
✨  android/app/src/main/res/values/colors.xml
✨  android/app/src/main/res/mipmap-mdpi/bootsplash_logo.png (288x288)
✨  android/app/src/main/res/mipmap-xxhdpi/bootsplash_logo.png (864x864)
✨  android/app/src/main/res/mipmap-xhdpi/bootsplash_logo.png (576x576)
✨  android/app/src/main/res/mipmap-xxxhdpi/bootsplash_logo.png (1152x1152)
✨  android/app/src/main/res/mipmap-hdpi/bootsplash_logo.png (432x432)

    iOS
✨  ios/MunchGo/BootSplash.storyboard
✨  ios/MunchGo/Images.xcassets/BootSplashLogo.imageset/bootsplash_logo.png (288x624)
✨  ios/MunchGo/Images.xcassets/BootSplashLogo.imageset/bootsplash_logo@2x.png (576x1248)
✨  ios/MunchGo/Images.xcassets/BootSplashLogo.imageset/bootsplash_logo@3x.png (864x1871)

 ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
 ┃  💖  Love this library? Consider sponsoring!  ┃
 ┃  One-time amounts are available.              ┃
 ┃  https://github.com/sponsors/zoontek          ┃
 ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

✅  Done! Thanks for using react-native-bootsplash.

But my screen still doesn't fit with the splash screen.
Please take a look my image:
<img width="420" alt="image" src="https://user-images.githubusercontent.com/12009276/201151325-e1877b53-791d-45d9-b2b1-37738b2f3d5b.png">

So how do i make it fit with splash screen?
thank you so much

### Library version

^4.3.2

### Environment info

```shell
System:
    OS: macOS 12.4
    CPU: (8) x64 Apple M1 Pro
    Memory: 24.61 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.19.0 - ~/.nvm/versions/node/v14.19.0/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v14.19.0/bin/yarn
    npm: 7.9.0 - ~/.nvm/versions/node/v14.19.0/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 30, 31, 32, 33
      Build Tools: 30.0.2, 30.0.3, 31.0.0, 32.0.0, 32.1.0
      System Images: android-32 | Google APIs ARM 64 v8a, android-Tiramisu | Google Play ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8193401
    Xcode: 13.4/13F17a - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.15.1 - /Library/Java/JavaVirtualMachines/jdk-11.0.15.1.jdk/Contents/Home/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0 
    react-native: 0.70.1 => 0.70.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

I used the default command without logo-width: yarn react-native generate-bootsplash assets/bootsplash_logo_original.png
But i got the small image on the splash screen. And after that i use --logo-width=300, and it got the error: ` Logo width can't be superior to 288dp as it will be cropped on Android. Exiting…`
Then i changed to `npx react-native generate-bootsplash ./assets/images/splash.png \ --background-color=FFFFFF \
  --logo-width=288`

### Reproducible sample code

```js
.
```

**Comment by zoontek** [comment]
It's in dpi (when @1x 1dpi = 1px, exactly like in React Native)

Android have a hard limitation of a circle mask, of 288x288dpi.

- Which means if your logo width is > 288dpi, it will be cropped, 100% sure.
- If your logo width is 288dpi, but is a perfect round, it will not be cropped.
- If your logo width is < 288dpi, but 192dpi, it might be cropped depending on its shape (how it fits in the circle mask)
- If your logo width is < 192dpi, it cannot be cropped, because it fits in the square inside the circle mask:

![1DF83FA4-E82F-4864-A858-A8AF6B5C8112](https://user-images.githubusercontent.com/1902323/201156741-93b37bd7-40e4-4f6a-ae1e-efbe38f13394.jpeg)

More infos: https://developer.android.com/develop/ui/views/launch/splash-screen#elements

**Comment by hungdev** [comment]
@zoontek thank you for replying. But I'm trying to install it on IOS. And ios it's not fit. This image is ios platform.
<img width="420" alt="image" src="https://user-images.githubusercontent.com/12009276/201151325-e1877b53-791d-45d9-b2b1-37738b2f3d5b.png">

**Comment by zoontek** [comment]
You never mentioned iOS in your issue.

Designing your splash screen is not limited on iOS, you can use Xcode storyboard editor to adjust it the way you want:

<img width="1499" alt="Screenshot 2022-11-11 at 11 39 20" src="https://user-images.githubusercontent.com/1902323/201323428-16c7f8da-f8e1-4efd-9835-74abd081bd0f.png">

You can search online for tutorials about image constraints, etc.

**Comment by brianlenz** [comment]
@zoontek sorry to rehash an old issue, but are you sure your logo size assessment is correct? We tried cropping our logo to 192x139. We didn't get a warning with the `generate` command, but the logo was still cropped. Then I found this image on the [Android documentation](https://developer.android.com/develop/ui/views/launch/splash-screen#dimensions):

<img width="750" alt="Image" src="https://github.com/user-attachments/assets/defdbd9f-3570-435e-a55c-6770acd22196" />

This seems to indicate that the _circle_ diameter is 192px, in which case you'd have to fit the logo inside a square _within_ that circle to avoid cropping, which reduces the size to about 134px per edge. When we reduce our logo to 134x97, _then_ it is not cropped in Android.

Am I missing something? Thanks, much 🙏
